src/GraphBLAS-sharp.Backend/MatrixOperations/COOMatrix/EWiseAddBatched.fs

```diff
@@ -0,0 +1,507 @@
+            @>
+
+        let kernel = clContext.Compile(merge)
+
+        fun (processor: MailboxProcessor<_>)
+            (matrixLeftRows: ClArray<int>)
+            (matrixLeftColumns: ClArray<int>)
+            (matrixLeftValues: ClArray<'a>)
+            (matrixRightRows: ClArray<int>)
+            (matrixRightColumns: ClArray<int>)
+            (matrixRightValues: ClArray<'a>) ->
+
+            let firstSide = matrixLeftValues.Length
+            let secondSide = matrixRightValues.Length
+            let sumOfSides = firstSide + secondSide
+
+            let allRows =
+                clContext.CreateClArray<int>(
+                    sumOfSides,
+                    deviceAccessMode = DeviceAccessMode.WriteOnly,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let allColumns =
+                clContext.CreateClArray<int>(
+                    sumOfSides,
+                    deviceAccessMode = DeviceAccessMode.WriteOnly,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let allValues =
+                clContext.CreateClArray<'a>(
+                    sumOfSides,
+                    deviceAccessMode = DeviceAccessMode.WriteOnly,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    allocationMode = AllocationMode.Default
+                )
+
+
+            let ndRange = Range1D.CreateValid(sumOfSides, workGroupSize)
+            let kernel = kernel.GetKernel()
+
+            processor.Post(
+                Msg.MsgSetArguments(fun () ->
+                    kernel.KernelFunc
+                        ndRange
+                        firstSide
+                        secondSide
+                        sumOfSides
+                        matrixLeftRows
+                        matrixLeftColumns
+                        matrixLeftValues
+                        matrixRightRows
+                        matrixRightColumns
+                        matrixRightValues
+                        allRows
+                        allColumns
+                        allValues
+                )
+            )
+
+            processor.Post(Msg.CreateRunMsg<_, _>(kernel))
+
+            allRows, allColumns, allValues
+
+    let combineIndices (clContext: ClContext) workGroupSize =
+        let command =
+            <@
+                fun (range: Range1D)
+                    (rowsBuffer: ClArray<int>)
+                    (colsBuffer: ClArray<int>)
+                    (indicesBufferLength: int)
+                    (colsCount: int)
+                    (newIndecies: ClArray<int>) ->
+
+                    let gid = range.GlobalID0
+
+                    if gid < indicesBufferLength then
+                        newIndecies.[gid] <- rowsBuffer.[gid] * colsCount + colsBuffer.[gid]
```

Comment:
This mapping is not injective. Although, this can cause improper behavior in extremely rare cases